Add SOP0011 for soft withdrawal from node FAIR Data Point (#71)
GenomicDataInfrastructure/standard-operating-procedures: sops/node-specific/GDI-SOP0011_withdraw-dataset-from-node-fair-data-point.md

| URI | Uniform Resource Identifier |
| VHD | Virtual Helpdesk |

| Term | Definition |
| -- | -- |
| Distribution | A specific access representation of a dataset in the FAIR Data Point, such as a download, landing page, or service endpoint. |
| FAIR Data Point | The node-managed metadata catalogue used to publish and maintain dataset and distribution descriptions. |
| HealthDCAT-AP | The health-focused application profile used to structure dataset metadata, including controlled status values. |
| Tombstone | A minimal metadata record that remains visible after withdrawal to preserve traceability and indicate that the dataset or distribution has been withdrawn. |
| Withdrawal | A lifecycle state indicating that a dataset or distribution is no longer offered for normal use through the FAIR Data Point. |

### 3. Roles and Responsibilities

See qualifications and responsibilities of the roles at the [**Organisational Roles and Responsibilities**](../../docs/GDI-SOP_organisational-roles-and-responsibilities.md) document.

| Role | Full name | GDI/node role | Organisation |
| -- | -- | -- | -- |
| Author | Hans-Christian van der Werf | SOP author and Task 4.3 contributor | Health-RI |

> | Reviewer | Marcos Casado Barbero | Task 4.3 lead | EMBL-EBI |

Reviewer: ```suggestion
| Reviewer | Marcos Casado Barbero | Task 4.3 member | EMBL-EBI |
```
Not leading it anymore 😉

Author: Updated. I changed the reviewer row to `Task 4.3 member`, and I also filled the temporary approver/authorizer rows following the current repo conventions.

| Approver | To be assigned | Approver according to GDI SOP governance | To be assigned |
| Authorizer | To be assigned | Authorizer according to GDI SOP governance | To be assigned |

### 4. Purpose

> This SOP defines how a GDI node checks whether a dataset requested for withdrawal is present in its FAIR Data Point (FDP), determines the withdrawal scope, and performs a soft withdrawal in a consistent and auditable way. Withdrawal is implemented by setting `adms:status` on the dataset and/or affected distributions while keeping a tombstone record where applicable for traceability.

Reviewer: If I remember correctly from the original SOP request, and the ``SOP0009`` on dataset withdrawal, the scope is determined by the data controller. Therefore, I wouldn't say we are determining it here in the SOP, but rather following the users' request. Same with the assumption that all of them will be ``soft`` deletions (?)

Reviewer: Furthermore, isn't ``adms:status`` a ``Distribution`` level attribute in HealthDCAT-AP (v5 and v6)? Are FDP records not catalogue records or Datasets, as opposed to distributions of the dataset? Or am I misunderstanding the entity types?

Author: I updated the SOP to make this narrower and more explicit. It now says the FDP maintainer follows the approved/requested scope coming from SOP0009 rather than determining it locally, and the PR title/body plus issue #63 comment now explicitly document that this PR only covers the FDP soft-withdrawal subset.

Author: I clarified this in the background section. The SOP now states distribution-level `adms:status` as the HealthDCAT-AP / EU distribution-status usage, while dataset-level `adms:status` is described as a GDI metadata-profile convention aligned with Health-RI and the GDI metadata profile, not as a direct HealthDCAT-AP requirement.

| EMBL | European Molecular Biology Laboratory |
| FAIR | Findability, Accessibility, Interoperability and Reusability |
| FDP | FAIR Data Point |
| GDI | European Genomic Data Infrastructure |
| ISM | Information Service Management |
| ORR | Organisational Roles and Responsibilities |
| PID | Permanent Identifier |
| PR | Pull Request |
| RI | Research Infrastructure |
| SOP | Standard Operating Procedure |
| TB | Top to Bottom |
| VHD | Virtual Helpdesk |

| Term | Definition |
| -- | -- |
| Withdrawal | A lifecycle state indicating that a dataset or distribution is no longer offered for normal use through the FAIR Data Point. |

### 3. Roles and Responsibilities

See qualifications and responsibilities of the roles at the [**Organisational Roles and Responsibilities**](../../docs/GDI-SOP_organisational-roles-and-responsibilities.md) document.

| Role | Full name | GDI/node role | Organisation |
| -- | -- | -- | -- |
| Author | Hans-Christian van der Werf | SOP author and Task 4.3 contributor | Health-RI |
| Reviewer | Marcos Casado Barbero | Task 4.3 lead | EMBL-EBI |
| Approver | To be assigned | Approver according to GDI SOP governance | To be assigned |

> | Authorizer | To be assigned | Authorizer according to GDI SOP governance | To be assigned |

Reviewer: I would suggest you add @GabiRinck as an Approver for now, until (if) the OC/SDPC elects another.

Also, for authorizer, I would write the management board, who will have the 30 day veto period to give feedback.

Author: Applied. I set Gabriele Rinck as Approver and Management Board as Authorizer in the roles table.

### 4. Purpose

This SOP defines how a GDI node checks whether a dataset requested for withdrawal is present in its FAIR Data Point (FDP), determines the withdrawal scope, and performs a soft withdrawal in a consistent and auditable way. Withdrawal is implemented by setting `adms:status` on the dataset and/or affected distributions while keeping a tombstone record where applicable for traceability.

### 5. Scope

> This SOP covers the processing of a soft withdrawal request in a Node FAIR Data Point. It includes locating the dataset record, identifying linked distributions, determining whether the withdrawal is full or distribution-specific, applying soft withdrawal through `adms:status`, verifying the resulting visibility of the withdrawn records in the FDP, logging the action, and reporting completion to the GDI Virtual Helpdesk. It assumes that the central user portal harvests updated metadata automatically and therefore does not require a separate manual re-indexing action there.

Reviewer: I would say 90% of the deletions may be soft deletions, but assuming that they will in the SOP is misleading. 

I'm not familiar with FDP myself, but I see that its reference API exposes a ``DELETE`` on resource endpoints like ``dataset``. Would that not be the route for hard-deletion?

Instead, I would phrase it as this SOP encompassing **both** hard and soft deletions. Also, to make it simpler, I would say that "unless the original request explicitly mentions a hard deletion, all deletions would go through the soft route", to be FAIR.

Author: I kept the SOP intentionally narrow rather than broadening it to hard deletion in this PR. To make that explicit, I updated the PR title/body, narrowed the SOP purpose/scope wording, and added an issue comment on #63 listing what is covered and what remains out of scope.

> This SOP covers the processing of a soft withdrawal request in a Node FAIR Data Point. It includes locating the dataset record, identifying linked distributions, determining whether the withdrawal is full or distribution-specific, applying soft withdrawal through `adms:status`, verifying the resulting visibility of the withdrawn records in the FDP, logging the action, and reporting completion to the GDI Virtual Helpdesk. It assumes that the central user portal harvests updated metadata automatically and therefore does not require a separate manual re-indexing action there.

Reviewer: Since we're assuming that the central user portal harvests and updates automatically, I would add to the validation step of this SOP **explicitly checking the central portal**, to make sure it's updated.

Author: Added. Step 8.5 now keeps the FDP as the immediate verification point and also says the central portal should be checked after automatic harvest when that is operationally relevant.

> A full withdrawal means that the dataset and all linked distributions are marked as withdrawn. A distribution-specific withdrawal means that one or more affected distributions are marked as withdrawn while the dataset record itself remains active. This SOP does not cover hard deletion, deletion of individual metadata elements within a dataset record, participant-level deletion inside a dataset, or downstream withdrawal activities in other systems unless those are defined in another SOP.

Reviewer: ```suggestion
- A full withdrawal means that the dataset and all linked distributions are marked as withdrawn. 
- A distribution-specific withdrawal means that one or more affected distributions are marked as withdrawn while the dataset record itself remains active. 
 

This SOP **does not** cover hard deletion, deletion of individual metadata elements within a dataset record, participant-level deletion inside a dataset, or downstream withdrawal activities in other systems unless explicitly mentioned.
```
Regarding the scope:
- As mentioned above, we would want to cover hard deletion. We could start nonetheless with a soft-only SOP, but have the idea to update it in the future to include the other type.
- Initially we were hoping to get the ``partial`` part of deletions to get into granularity of individual records or elements; since this is not the case, I would recommend to add a brief sub-section below simply explaining something along the lines of "in the FDP the minimal removal unit is dataset/distribution as a whole..." or similar

### 6. Introduction and Background Information

A node's FAIR Data Point is often the first public-facing catalogue where a dataset becomes discoverable. Without a standard withdrawal procedure, datasets or distributions that should no longer be offered may remain visible or accessible, undermining governance, compliance, and user trust. This SOP ensures that nodes apply a consistent and auditable soft withdrawal process in the FDP while keeping a tombstone where appropriate.

In this SOP, withdrawal is expressed using `adms:status`. At dataset level, use the European Union Dataset Status vocabulary and set the value to [http://publications.europa.eu/resource/authority/dataset-status/withdrawn](http://publications.europa.eu/resource/authority/dataset-status/withdrawn). At distribution level, use the European Union Distribution Status vocabulary and set the value to [http://publications.europa.eu/resource/authority/distribution-status/withdrawn](http://publications.europa.eu/resource/authority/distribution-status/withdrawn).

> For a full withdrawal, set the dataset and all linked distributions to `Withdrawn`. For a distribution-specific withdrawal, set only the affected distributions to `Withdrawn`. Withdrawn records should remain visible as tombstones when the node implementation or governance model requires traceability.

Reviewer: ```suggestion
For a **full withdrawal**, set the dataset and all linked distributions to `Withdrawn`. For a **distribution-specific withdrawal**, set only the affected distributions to `Withdrawn`. Withdrawn records should remain visible as tombstones when the node implementation or governance model requires traceability.
```

Author: Updated in that direction. The paragraph now keeps the full vs distribution-specific distinction explicit and leaves the tombstone behaviour as conditional on node implementation or governance requirements.

### 7. Summary or Context Diagram

```mermaid
flowchart TB
    request["Approved withdrawal request"]
    find["Locate dataset record in FDP"]
    dist["Identify linked distributions"]
    scope{"Full or distribution-specific withdrawal?"}
    full["Set dataset and all distributions to Withdrawn"]
    partial["Set affected distributions to Withdrawn"]
    save["Save updated records in FDP"]
    verify["Verify FDP visibility and tombstone"]
    harvest["Central portal harvests updated metadata automatically"]
    log["Log action"]
    report["Report completion in the withdrawal request"]

    request --> find --> dist --> scope
    scope -->|Full| full --> save
    scope -->|Distribution-specific| partial --> save
    save --> verify --> harvest --> log --> report
```

### 8. Procedure

#### 8.1. Confirm withdrawal request and identify dataset

| Step identifier | When | Who |
|:--|:--|:--|

> | 1 | When a dataset withdrawal has been approved or formally requested under the overarching withdrawal process described in issue [#55](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/issues/55) and pull request [#62](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/pull/62). | Node FAIR Data Point maintainer or designated metadata curator |

Reviewer: ```suggestion
| 1 | When a dataset withdrawal `request has been approved under the overarching withdrawal process described in [``GDI-SOP0009_dataset-withdrawal.md``](../european-level/GDI-SOP0009_dataset-withdrawal.md#87-per-system-dataset-withdrawal)| Node FAIR Data Point maintainer or designated metadata curator |
```

Author: Updated. Step 8.1 now references SOP0009 directly by relative path and frames this as an already approved withdrawal request, not as a fresh authority check.

> As the node FAIR Data Point maintainer or designated metadata curator, confirm that the withdrawal request is valid and approved before changing metadata in the FAIR Data Point. Record the dataset identifier to be used as the primary lookup key. Record the dataset title only as an additional verification check.

Reviewer: I would add more details to what ``valid request`` means here: the intention is not for this role to check the validity of the request from an authoritative standpoint (i.e., whether the original requestor has authority)-- that is done by the original VHD officer in the SOP0009--, but instead to check that the email that is received by the VHD is valid. For example:
- It contains the original request
- It contains the scope of the withdrawal
- It contains the pertaining identifiers of the records to withdraw

Reviewer: Furthermore, based on this step's title and workflow, I would say that there needs to be a confirmation of request received to the VHD that sent it.

Author: Expanded. Step 8.1 now says to verify that the incoming request package is complete and explicitly lists the expected contents: the approved request/ticket reference, the requested scope, and the relevant identifiers. It also adds the clarification/pause branch if information is missing.

> As the node FAIR Data Point maintainer or designated metadata curator, confirm that the withdrawal request is valid and approved before changing metadata in the FAIR Data Point. Record the dataset identifier to be used as the primary lookup key. Record the dataset title only as an additional verification check.

Reviewer: Commonly in SOPs you want to be concise with the formatting, as helpdesk/operations will skim through it. For that reason I would suggest using the **format** of Markdown to our advantage, especially using bullet points, bold format and even emojis (e.g., ⏩).

Also, the SOP should be as **encompassing and explicit as possible**. For example, if this step says "check validity", the moment that there's a dichotomy in the workflow, the resulting routes should be explicitly mentioned. For example:
- ...if valid, go to step X(step X)
- ...if invalid, tell VHD ... or go to step Y(step Y) ... end of SOP

#### 8.2. Check presence of the dataset in the FAIR Data Point

| Step identifier | When | Who |
|:--|:--|:--|

> | 2 | After step 1. | Node FAIR Data Point maintainer or designated metadata curator |

Reviewer: ```suggestion
| 2 | After valid request at [Step 1][#81-confirm-withdrawal-request-and-identify-dataset] | Node FAIR Data Point maintainer or designated metadata curator |
```

Author: Applied. Step 2 now says it starts after successful completion of Step 1 and links back to that step directly.

> Search the FAIR Data Point for the dataset using the recorded identifier as the primary lookup key. Use the dataset title only as an additional check that the correct record has been found. If the dataset is found, proceed to the linked distributions. If the dataset is not found, record that outcome in the audit log and report the result back through the corresponding withdrawal request or VHD ticket.

Reviewer: ```suggestion
**Search** the FAIR Data Point for the dataset using the recorded **identifier** as the primary lookup key. Use the dataset title only as an additional check that the correct record has been found. 

- If the **dataset is found**, proceed to ⏩[Step 3](#83-identify-linked-distributions-and-assess-withdrawal-scope). 
- If the **dataset is not found**, record that outcome in the audit log and report the result back through the corresponding withdrawal request or VHD ticket.
```

Reviewer: An example of how I would recommend to format these steps. Not set in stone though.

Author: Applied in that direction. Steps 8.1 to 8.4 now use more explicit outcome bullets and step-to-step transitions so the skim path is clearer.

#### 8.3. Identify linked distributions and assess withdrawal scope

| Step identifier | When | Who |
|:--|:--|:--|
| 3 | After step 2. | Node FAIR Data Point maintainer or designated metadata curator |

Identify all linked distributions associated with the dataset, including service endpoints, landing pages, and download links where applicable. Determine whether the request concerns:

- a full withdrawal, meaning the dataset and all linked distributions
- a distribution-specific withdrawal, meaning only one or more affected distributions

> Only distribution-level partial withdrawal is covered by this SOP.

Reviewer: Why? Also, this was not in the scope section above, right? It seems in conflict with the general scope section, where we say "maybe partial, maybe full..."

Reviewer: Besides, if we're changing the status of entities, it seems fairly easy to change the status of one distribution or 10 of them along a dataset, right?

Reviewer: Reading now the step 8.4, I have the feeling this statement about partial-only was not meant.

Author: I removed that misleading sentence. The SOP now keeps distribution-specific withdrawal in scope and instead adds a short clarification that, in this FDP workflow, the minimum operational unit is the dataset record or a full linked distribution record rather than arbitrary metadata elements inside a record.

> 

Reviewer: I would add something again regarding "go to step x" at the end of each step. it may be that you're encompassing dichotomies as a single step (e.g., for full and partial), but the more explicit we are, the better.

Author: Kept the explicit step-routing pattern. Steps 1 to 4 now all include the next-step routing where it helps skimming, and step 3 already routes explicitly to Step 4 after the affected records have been identified.

#### 8.4. Apply soft withdrawal status

| Step identifier | When | Who |
|:--|:--|:--|
| 4 | After step 3. | Node FAIR Data Point maintainer or designated metadata curator |

If the withdrawal is full:

- open the dataset record in the FAIR Data Point editor

> - set dataset `adms:status` to `Withdrawn`

Reviewer: Shouldn't we be using the ``...WITHDRAWN`` URL here, as it's the value expected for the status of dataset/distribution based on the introduction above?

Author: Updated. Step 8.4 now says to set dataset/distribution `adms:status` to the `WITHDRAWN` value described above, instead of only naming the label.

- save the dataset record
- open each linked distribution
- set distribution `adms:status` to `Withdrawn`
- save each distribution record

If the withdrawal is distribution-specific:

- keep the dataset record unchanged
- open each affected distribution
- set distribution `adms:status` to `Withdrawn`
- save each affected distribution record

> Example operational instruction: go to FAIR Data Point, open the relevant record, change `Status` to `Withdrawn`, and press Save.

Reviewer: ??

Author: This was the overly terse verification/completion wording. I expanded that part so step 8.5 now covers FDP verification, optional central-portal follow-up after harvest, local logging, and reporting completion back to the VHD.

#### 8.5. Verify, log, and report completion

| Step identifier | When | Who |
|:--|:--|:--|
| 5 | After step 4. | Node FAIR Data Point maintainer or designated metadata curator |

> Verify in the FAIR Data Point that the withdrawal result matches the requested scope:

Reviewer: I would also add the central portal, or else the assumption above is risky.

Author: Added. Step 8.5 now keeps the node FDP as the immediate verification point and also includes the central portal as a follow-up check after automatic harvest when that is operationally relevant.

- full withdrawal: the dataset and all linked distributions are marked `Withdrawn`
- distribution-specific withdrawal: only the affected distributions are marked `Withdrawn`

> Verify that withdrawn records remain visible as tombstones if this is GDI catalogue behaviour.

Reviewer: Are FDP records with ``Withdrawn`` still fully visible (i.e., their metadata)? In soft deletion, one can have a tombstone that is an empty record, with the title and little more, or a full record. Some data controllers will not want the deletion to simply leave all metadata hanging, and it seems we're not covering hard deletion.

For example, if a data controller needs to remove all linkage to a dataset for privacy issues, leaving fully detailed tombstones may not be in their scop.e

Author: I tightened that wording. The SOP now says tombstones remain visible only if that is the expected GDI catalogue behaviour for the affected record type, which keeps it compatible with the narrow soft-withdrawal scope without claiming that full metadata must remain visible in every case.

> Report completion in the requestor.

Reviewer: I would recommend adding details on how to report completion: who/when/what/scope/reason... 

Furthermore, it should **not** be to requester, but to the VHD. All communication goes through VHD.

Author: Changed. Step 8.5 now reports completion back to the GDI Virtual Helpdesk and adds explicit local logging details. Requester communication is no longer described as happening directly from this SOP.

| ---------- | --------------------------- | ------------------------------------------------ | ------------------------------------- |
| Author     | Hans-Christian van der Werf | SOP author and Task 4.3 contributor              | Health-RI                             |
| Reviewer   | Marcos Casado Barbero       | Task 4.3 member                                  | EMBL-EBI                              |
| Approver   | Gabriele Rinck              | Task 4.3 member                                  | EMBL-EBI                              |
| Authorizer | Management Board            | Authorizer according to GDI SOP governance       | GDI                                   |

### 4. Purpose

This SOP defines how a GDI node checks whether a dataset requested for withdrawal is present in its FAIR Data Point (FDP), follows the approved withdrawal scope received through the overarching dataset-withdrawal process, and performs a soft withdrawal in a consistent and auditable way. Withdrawal is implemented by setting `adms:status` on the dataset and/or affected distributions while keeping a tombstone record where applicable for traceability.

### 5. Scope

> This SOP covers the processing of an approved soft withdrawal request in a Node FAIR Data Point. It includes locating the dataset record, identifying linked distributions, following the requested withdrawal scope, applying soft withdrawal through `adms:status`, verifying the resulting visibility of the withdrawn records in the FDP, logging the action, and reporting completion to the GDI Virtual Helpdesk. It assumes that the central user portal harvests updated metadata automatically and therefore does not require a separate manual re-indexing action there.

Reviewer 2: The term `Node FAIR Data Point` is used inconsistently, appearing in some places as `node FAIR Data Point`. It would be great to make them consistent.

Author: Standardized. The SOP now consistently uses `Node FAIR Data Point` when referring to the template context, and keeps possessive wording only where grammar requires `nodes FAIR Data Point`.

| Abbreviation | Description |
| -- | -- |

> | API | Application Programming Interface |

Reviewer: You removed this one from the text

Author: Adjusted. I reduced the glossary again and removed the earlier entries that were no longer needed after the wording cleanup.

| EBI | European Bioinformatics Institute |
| EMBL | European Molecular Biology Laboratory |
| FAIR | Findability, Accessibility, Interoperability and Reusability |
| FDP | FAIR Data Point |
| GDI | European Genomic Data Infrastructure |
| ISM | Information Service Management |
| ORR | Organisational Roles and Responsibilities |

> | PID | Permanent Identifier |

Reviewer: Doesn't appear in the text either

Reviewer: There may be more, as I just checked 2-3

Author: I reduced the glossary again. The remaining entries are there because they now appear directly in the text or in reviewer-requested wording, but I dropped the earlier filler additions that were only there to satisfy the linter.

Author: Tightened further. I removed the unused `PID` and `RI` glossary entries as well, so the remaining glossary items are only the ones still used directly in the SOP text or role labels.

A node's FAIR Data Point is often the first public-facing catalogue where a dataset becomes discoverable. Without a standard withdrawal procedure, datasets or distributions that should no longer be offered may remain visible or accessible, undermining governance, compliance, and user trust. This SOP ensures that nodes apply a consistent and auditable soft withdrawal process in the FDP while keeping a tombstone where appropriate.

> In this SOP, withdrawal is expressed using `adms:status`. At dataset level, use the European Union Dataset Status vocabulary and set the value to [http://publications.europa.eu/resource/authority/dataset-status/withdrawn](http://publications.europa.eu/resource/authority/dataset-status/withdrawn). At distribution level, use the European Union Distribution Status vocabulary and set the value to [http://publications.europa.eu/resource/authority/distribution-status/withdrawn](http://publications.europa.eu/resource/authority/distribution-status/withdrawn).

Reviewer: The publications.europa.eu link is caps-sensitive, I believe. This URL doesn't lead to the correct definition, as opposed to the one in caps.

Author: Updated. I restored the canonical uppercase Publications Office URLs in the references and in the explanatory status paragraph.
